test/net/sourceforge/kolmafia/textui/command/ItemTraceCommandTest.java

```diff
@@ -0,0 +1,59 @@
+package net.sourceforge.kolmafia.textui.command;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.core.StringContains.containsString;
+
+import internal.helpers.Cleanups;
+import internal.helpers.Player;
+import internal.helpers.RequestLoggerOutput;
+import org.junit.jupiter.api.Test;
+
+public class ItemTraceCommandTest extends AbstractCommandTestBase {
+
+  public ItemTraceCommandTest() {
+    this.command = "itrace";
+  }
+
+  @Test
+  public void doesNotTraceFakeItems() {
+    execute("beefy nigiri");
+
+    RequestLoggerOutput.startStream();
+    Cleanups cleanups = Player.addItem("beefy nigiri");
+    try (cleanups) {
+      var text = RequestLoggerOutput.stopStream();
+      assertThat(text, not(containsString("itrace")));
+    }
+
+    String output = execute("");
+    assertThat(output, containsString("Previously watched items have been cleared"));
+  }
```

Comment:
This cleans up item listener at end of test.

Reply:
This is to cover the case where an item listener is `unregister`ed without having been registered in the first place -- which happens if the item added has a negative id (i.e. is not a real item).